Add building of RPMs and update default paths for DEB packages and command line parameters

soenkeliebau opened the pull request:
This PR makes a number of changes to the agent for both RPM and DEB packages, as well as default values for command line options.

I would have liked to split this into separate PRs, but as all these things are tied together there was no clean way of separating this, so I decided to bunch everything together.

Agent Code:
In line with [ADR011](https://github.com/stackabletech/documentation/blob/main/adr/ADR011-directory_structure.adoc) default value for the following parameters was changed (old value in [] for reference).
- data-directory: /var/lib/stackable/agent [/var/stackable/agent/data]
- bootstrap-file: /etc/stackable/stackable-agent/bootstrap-kubelet.conf [/etc/kubernetes/bootstrap-kubelet.conf]
- server-cert-file: /etc/stackable/stackable-agent/secret/agent.crt [None]
- server-key-file: /etc/stackable/stackable-agent/secret/agent.key [None]
- config-directory: /etc/stackable/serviceconfig [/opt/stackable/config]
- log-directory: /var/log/stackable/servicelogs [/opt/stackable/logs]

RPM:
Converted build process from operators to build agent rpm.

DEB:
Changed paths to be in line with the new defaults specified above.

Both packages currently do not deliver a config file, as they run completely of the default values. We should look into packaging a template config perhaps.

I am not entirely happy with the current directory structure for packaging, as we have some duplication in here, but I think it will serve for now and is not worth the effort needed to make it nicer (which would also make it more complex).

Also, changes to paths which files are installed into currently require a bit of digging and changes in multiple files - again, this is not nice, but I'm not sure it is worth the effort of making it nicer.

I've created issue #108 with some open questions that do not necessarily need to be included in this PR but should be addressed at some point.

soenkeliebau committed:
First try of building and pushing an rpm package.

soenkeliebau committed:
Added debug flags to rpm build process and forgotten config file.

soenkeliebau committed:
Fixes for rpm publishing process.

soenkeliebau committed:
Removed leftover backslashes from testing this in shell script.
Added directory creation

soenkeliebau committed:
Removed test trigger to run bild on push to rpm branch.

backstreetkiwi commented:
My review applies not to the way the packages are built here. Instead, as a user of the built artifacts I can say that the CentOS packages that are built by the action in this branch work in my tests with cluster generation in T2.

siegfriedweber commented:
The paths described in the [documentation](https://github.com/stackabletech/agent/tree/764abb482dda7e699f731130d4faac916304ca2b) do not match with the ones configured in the rpm.

soenkeliebau committed:
Merge branch 'main' into rpm

# Conflicts:
#	Cargo.toml

soenkeliebau committed:
Ported build code from operators for rpm.

Also changed default paths for a lot of agent directories - this definitely needs calling out in the commit message!

Config file is now empty, as all settings are used with their default values.

soenkeliebau committed:
Typo

soenkeliebau committed:
Forgot to commit specfile

soenkeliebau committed:
Directory naming error

soenkeliebau committed:
Config File was not specified as managed.

soenkeliebau committed:
Typo in conf dir

soenkeliebau committed:
Fixed name of binary in apt spec

soenkeliebau committed:
Fixed service file for debian, config path was not adjusted in here.

soenkeliebau committed:
Fixed a mixture of errors in config paths.

soenkeliebau committed:
Duplication in path

soenkeliebau committed:
Changed default value for tls files path

soenkeliebau committed:
Fixed path

soenkeliebau committed:
Removed running build on rpm branch in preparation for merging

soenkeliebau committed:
Regenerated command line parameter docs.

lfrancke commented on .rpm/systemd/stackable-agent.service:
```diff
@@ -0,0 +1,14 @@
+Restart=on-abort
+StandardOutput=journal
+StandardError=journal
+Environment="CONFIG_FILE=/etc/stackable-agent/agent.conf"
```
Is this correct?

soenkeliebau replied:
File is obsolete.

lfrancke commented on packaging/rpm/SPECS/stackable-agent.spec:
```diff
@@ -0,0 +1,62 @@
+%define __spec_install_post %{nil}
```
there are two stackable-agent.spec files. Which one is correct or are they both used?

soenkeliebau replied:
Removed the incorrect one.

lfrancke reviewed (commented):
I think I have found issues but I'm not 100% sure. Please double check.

lfrancke reviewed (changes requested):
See previous review

soenkeliebau committed:
Merge branch 'main' into rpm

soenkeliebau commented on .rpm/stackable-agent.spec:
```diff
@@ -0,0 +1,57 @@
+%define __os_install_post %{_dbpath}/brp-compress
+%define debug_package %{nil}
+%define _bindir /opt/stackable-agent
+%define _confdir /etc/stackable-agent
```
This entire file is not needed anymore, I simply forgot to include deletion of the now unused .rpm folder in my commit.

soenkeliebau committed:
Adressed comments from Lars.

soenkeliebau commented:
I have addressed the comments

> The paths described in the [documentation](https://github.com/stackabletech/agent/tree/764abb482dda7e699f731130d4faac916304ca2b) do not match with the ones configured in the rpm.

I'm not entirely sure if this comment still needs addressing after all the path changes I included in this PR?
